client/src/components/MyProfile.js

```diff
@@ -1,9 +1,272 @@
-import React from 'react'
+import { React, useState, useEffect } from 'react';
+import '../styles/MyProfile.css';
+import Button from "@mui/material/Button";
+import EditDetailsModal from '../modals/EditDetailsModal';
+
+function EventItem(props)
+{
+
+  // These states store the original event data
+  const [Firstname, setFirstname] = useState(props.Firstname);
```

Comment:
I think these should not have capital first letters, same with the history variables  :P